attributes/default.rb

```diff
@@ -68,3 +68,57 @@
+
+# Set Log file for kafka init script stdout/stderr
+default["kafka"]["mirror_maker"]["service"]["stdout"] = File.join node["kafka"]["log_dir"], "kafka_mirror_maker_init_stdout.log"
+default["kafka"]["mirror_maker"]["service"]["stderr"] = File.join node["kafka"]["log_dir"], "kafka_mirror_maker_init_stderr.log"
+
+default["kafka"]["mirror_maker"]["env_vars"]["JMX_PORT"] = "9998"
+default["kafka"]["mirror_maker"]["env_vars"]["KAFKA_HEAP_OPTS"] = "\"-Xmx1G -Xms1G\""
+default["kafka"]["mirror_maker"]["env_vars"]["KAFKA_JVM_PERFORMANCE_OPTS"] = "\"-XX:PermSize=48m -XX:MaxPermSize=48m -XX:+UseG1GC -XX:MaxGCPauseMillis=20 -XX:InitiatingHeapOccupancyPercent=35\""
+
+# mirror maker Log4J config
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.rootLogger"] = "INFO, stdout "
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stdout"] = "org.apache.log4j.ConsoleAppender"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stdout.layout"] = "org.apache.log4j.PatternLayout"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stdout.layout.ConversionPattern"] = "[%d] %p %m (%c)%n"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.kafkaAppender"] = "org.apache.log4j.RollingFileAppender"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.kafkaAppender.MaxBackupIndex"] = "20"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.kafkaAppender.layout"] = "org.apache.log4j.PatternLayout"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.kafkaAppender.layout.ConversionPattern"] = "%d{ISO8601} %p %c: %m%n"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stateChangeAppender"] = "org.apache.log4j.RollingFileAppender"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stateChangeAppender.MaxBackupIndex"] = "20"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stateChangeAppender.layout"] = "org.apache.log4j.PatternLayout"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.stateChangeAppender.layout.ConversionPattern"] = "%d{ISO8601} %p %c: %m%n"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.requestAppender"] = "org.apache.log4j.RollingFileAppender"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.requestAppender.MaxBackupIndex"] = "20"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.requestAppender.layout"] = "org.apache.log4j.PatternLayout"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.requestAppender.layout.ConversionPattern"] = "%d{ISO8601} %p %c: %m%n"
+default["kafka"]['mirror_maker']["mirror-log4j.properties"]["log4j.appender.controllerAppender"] = "org.apache.log4j.RollingFileAppender"
```

Comment:
I don't think you need the state change, request or controller config. I'm pretty sure thats kafka broker specific